### PR TITLE
fix: wrong end.offset - close #279

### LINKED ISSUE #279: Parsing error when a component is used within a markdown list
### Subject of the issue

The following code:

```mdx
- <LinkTo kind="docs-packages-vuetify-preset" story="page">
    Vuetify preset
  </LinkTo>
  : some extra text describing the preset
```

Trigger an error on line 3:

```
  </LinkTo>
    ^^^^^^
    Parsing error: '>' expected.
```

Adding an empty line between the last two lines fixes the issue, but the rendered output is not the same. Also putting it all in a single line works, but it’s not "Prettier compliant" and gets reformatted on multiple lines. If the same code is placed at the root (not within a markdown list), it works as well.

It does pretty much the same with a regular element instead of a component, but the error is different:

```
Parsing error: Identifier expected
```

Note that the code compiles properly, the problem is only at EsLint level.

### Your environment

*   **OS**: macOS 10.15.7
*   **Packages**: eslint-plugin-mdx 1.9.0
*   **Env**: Node 14.16.0, npm 6.14.11

### PR DESCRIPTION
**Pull request by JounQin**
close #279

Don't know if it is a good fix. Or should it be fixed in `remark-mdx` instead.

### CODE REVIEW

**Comment by JounQin** [comment]
@wooorm I'd like to give it a try still.

File: packages/eslint-mdx/src/traverse.ts [hunk 1]
```diff
@@ -20,16 +20,23 @@ export class Traverse {
     this._enter = enter
   }
 
-  combineLeftJsxNodes(jsxNodes: Node[]) {
+  combineLeftJsxNodes(jsxNodes: Node[], parent?: Parent): Node {
     const start = jsxNodes[0].position.start
-    const end = last(jsxNodes).position.end
+    const end = { ...last(jsxNodes).position.end }
+    // fix #279
+    if (parent && parent.position.indent?.length > 0) {
```
Comment (JounQin) [inline packages/eslint-mdx/src/traverse.ts:27]: https://github.com/istanbuljs/istanbuljs/issues/526

**Comment by wooorm** [comment]
> I'd like to give it a try still.

If it works, it works! It looks like you’ve tested this well, so that’s a :+1: from me.

**Comment by JounQin** [comment]
@wooorm So it's ready for a patch release!

**Comment by wooorm** [comment]
released!

**Pull request merged**